shell/components/ResourceTable.vue

```diff
@@ -344,7 +365,12 @@ export default {
         const exists = this.groupOptions.find((g) => g.value === this._group);
 
         if (!exists) {
-          return DEFAULT_GROUP;
+          // Attempt to find the default option in available options...
+          // if not use the first value in the options collection...
+          // and if not that just fall back to the default
+          const defaultGroup = this.groupOptions.find((g) => g.value === DEFAULT_GROUP);
+
+          return defaultGroup ? DEFAULT_GROUP : this.groupOptions[0]?.value || DEFAULT_GROUP;
```

Comment:
Using operator short circuiting in the same line of a ternary is a little difficult to parse and  determine the intent.

I imagine another an additional assignment could obviate the need for the above comment and make the return line a little less ambiguous.

Reply:
Good point. Have updated to the following 

```
          // Attempt to find the default option in available options...
          // if not use the first value in the options collection...
          // and if not that just fall back to the default
          if (this.groupOptions.find((g) => g.value === DEFAULT_GROUP)) {
            return DEFAULT_GROUP;
          }

          return this.groupOptions[0]?.value || DEFAULT_GROUP;
```